src/core/LiterallyCanvas.coffee

```diff
@@ -125,6 +124,15 @@ module.exports = class LiterallyCanvas
 
   setShapesInProgress: (newVal) -> @_shapesInProgress = newVal
 
+  selectShape: (shape) ->
+    @selectedShape = shape
+    if @selectedShape?
+      @setShapesInProgress [@selectedShape]
+      @trigger('shapeSelected', { @selectedShape })
+    else
+      @setShapesInProgress []
+    @repaintLayer('main')
```

Comment:
Yeah, I see your point! I'll move the selection / selection box drawing to the tool.